### PR TITLE
Made SlashingPenalty a Governance Parameter

### LINKED ISSUE #152: Make Slashing penalty as a governance parameter
**Describe the bug**
As of now slashing penalty is hardcoded, we need to make it a gov parameter and reduce it,
as directly slashing to 0 is too strict.

### PR DESCRIPTION
**Pull request by GauravJain9**
Fixes #152
Added a gov param as "percentSlashPenalty" by which we can decide and update how much percentage of the staker should be slashed and updated the slash function accordingly.

### CODE REVIEW

**Comment by GauravJain9** [comment]
@SkandaBhat @abhishekvispute  @SamAg19 Look into this PR

File: contracts/Core/StakeManager.sol [hunk 1]
```diff
@@ -292,17 +292,18 @@ contract StakeManager is Initializable, ACL, StakeStorage {
...
     function slash (uint256 id, address bountyHunter, uint256 epoch) external onlyRole(parameters.getStakeModifierHash()) {
-        uint256 halfStake = stakers[id].stake/(2);
-        _setStakerStake(id, 0, "Slashed", epoch);
-        if (halfStake > 1) {
-            require(sch.transfer(bountyHunter, halfStake), "failed to transfer bounty");
-        }
-    } 
+      uint256 bountyReward = ((stakers[id].stake)*(parameters.percentSlashPenalty()))/(2*100);
+      uint256 Stake =  stakers[id].stake - ((stakers[id].stake*parameters.percentSlashPenalty())/100);
+      if (bountyReward > 1) {
```
Comment (abhishekvispute) [inline contracts/Core/StakeManager.sol:302]: IMO We don't need this if, what it solves I am not clear
There are only two corner cases it excludes

bountyReward ==1 > I am not sure why we are not awarding bounty in this case
bountyReward ==0  > Even if its zero, sch.transfer wont transfer anything 

What do you guys say?
Maybe we should create diff issue and sep PR for same

Comment (GauravJain9) [reply contracts/Core/StakeManager.sol:302]: if (bountyReward > 0 && bountReward<1)
I think may be it is handling that @abhishekvispute

Comment (SamAg19) [reply contracts/Core/StakeManager.sol:302]: I agree with @abhishekvispute here. I would also suggest we remove that if statement

**Comment by GauravJain9** [comment]
ok will remove this @SamAg19 @abhishekvispute

File: contracts/Core/StakeManager.sol [hunk 2]
```diff
@@ -292,17 +292,18 @@ contract StakeManager is Initializable, ACL, StakeStorage {
...
     /// @param id The ID of the staker who is penalised
     /// @param bountyHunter The address of the bounty hunter
     function slash (uint256 id, address bountyHunter, uint256 epoch) external onlyRole(parameters.getStakeModifierHash()) {
-        uint256 halfStake = stakers[id].stake/(2);
-        _setStakerStake(id, 0, "Slashed", epoch);
-        if (halfStake > 1) {
-            require(sch.transfer(bountyHunter, halfStake), "failed to transfer bounty");
-        }
-    } 
+      uint256 bountyReward = ((stakers[id].stake)*(parameters.percentSlashPenalty()))/(2*100);
```
Comment (SkandaBhat) [inline contracts/Core/StakeManager.sol:300]: We are doing similar computation twice here. Can this be optimised?

File: contracts/Core/StakeManager.sol (hunk 1 above)
Comment (SkandaBhat) [reply contracts/Core/StakeManager.sol:302]: +1

**Review by SkandaBhat — CHANGES_REQUESTED** [review]

File: contracts/Core/StakeManager.sol (hunk 1 above)
Comment (SkandaBhat) [reply contracts/Core/StakeManager.sol:302]: Recorded this in #162

File: contracts/Core/StakeManager.sol (hunk 2 above)
Comment (abhishekvispute) [reply contracts/Core/StakeManager.sol:300]: Yes first penalty should be calculated 
Then bountyReward  = penalty / 2

File: contracts/Core/StakeManager.sol (hunk 1 above)
Comment (SkandaBhat) [reply contracts/Core/StakeManager.sol:302]: if (bountyReward = 0) return;
// go ahead

Comment (GauravJain9) [reply contracts/Core/StakeManager.sol:302]: will update this at once after updating all the test cases regarding this.

**Pull request closed without merging**